File: solutions/java/bob/1/src/main/java/Bob.java
public final class Bob {
    public String hey(String input) {
        String trimmedInput = input.trim();

        if (trimmedInput.isEmpty())
            return "Fine. Be that way!";
        
        if (trimmedInput.toUpperCase() == trimmedInput && trimmedInput.chars().anyMatch(Character::isLetter)) {
            if (trimmedInput.endsWith("?"))
                return "Calm down, I know what I'm doing!";
            
            return "Whoa, chill out!";
        }
        if (trimmedInput.endsWith("?"))
            return "Sure.";

        return "Whatever.";
    }
}

File: solutions/kotlin/bob/1/src/main/kotlin/Bob.kt
object Bob {
    fun hey(input: String): String {
        val trimmedInput: String = input.trim()
        
        if (trimmedInput.isSilence())
            return "Fine. Be that way!"
        if (trimmedInput.isYell()) {
            if (input.isQuestion())
                return "Calm down, I know what I'm doing!"
            return "Whoa, chill out!"
        }
        if (trimmedInput.isQuestion())
            return "Sure."
        
        return "Whatever."
    }

    private fun String.isSilence() = this == ""
    private fun String.isYell() = any(Char::isLetter) && toUpperCase() == this
    private fun String.isQuestion() = this.last() == '?'
}
